common/util.go

```diff
@@ -229,6 +230,13 @@ func IsServiceNonRetryableError(err error) bool {
 		return false
 	}
 
+	if errors.Is(err, context.DeadlineExceeded) {
```

Comment:
I think this makes sense if the same context will be used in later trials. I am just not sure if this is the case for all the places where this util function is called. 

Also probably add a comment for this function so that caller knows this error will be treated as non-retryable?

Reply:
I see. To keep the consistency, should we remove the check on L245?

Reply:
I checked the reference to those call and made a refactor to get rid of redundant retry. 
Our retry is a little bit over redundant (engine/handler -> history client -> frontend client -> user go/java client), I hope this change make it better.